src/PKSim.Core/Services/HIDiseaseStateImplementation.cs

```diff
@@ -165,41 +139,52 @@ private void updateHematocrit(Individual individual, Action<ParameterUpdate> upd
       }
 
-      private void updateBloodFlows(Individual individual, Action<ParameterUpdate> updateParameterFunc)
+      private void updateBloodFlowsAndVolumes(Individual individual, Action<ParameterUpdate> updateParameterFunc)
       {
          var score = childPughScoreFor(individual);
          var organism = individual.Organism;
-         var updateBloodFlow = updateBloodFlowDef(updateParameterFunc, organism);
-
-         //PortalGF
+         var updateBloodFlowSpec = updateBloodFlowSpecDef(updateParameterFunc, organism);
+         var bloodFlow = bloodFlowDef(organism);
          var portalOrgans = new[] {STOMACH, SMALL_INTESTINE, LARGE_INTESTINE, SPLEEN, PANCREAS};
-         portalOrgans.Each(x => updateBloodFlow(x, _portalFlowScalingFactor[score]));
-
-         //Hepatic
-         updateBloodFlow(LIVER, _hepaticFlowScalingFactor[score]);
-
-         //Renal
-         updateBloodFlow(KIDNEY, _renalFlowScalingFactor[score]);
-
-         //other organs
          var otherOrgans = new[] {BONE, FAT, GONADS, HEART, MUSCLE, SKIN};
-         otherOrgans.Each(x => updateBloodFlow(x, _otherOrgansFlowScalingFactor[score]));
-      }
-
-      private void updateVolumes(Individual individual, Action<ParameterUpdate> updateParameterFunc)
-      {
-         var score = childPughScoreFor(individual);
-         var organism = individual.Organism;
-         var parameter = organism.Container(LIVER).Parameter(VOLUME);
-         updateParameterFunc(new(parameter, _hepaticVolumeScalingFactor[score]));
+         var volumeLiver = organism.Container(LIVER).Parameter(VOLUME);
+
+         //Sum of all healthy portal blood flow + Liver and kidney
+         var Q_portal = portalOrgans.Sum(bloodFlow);
+         var Q_other = otherOrgans.Sum(bloodFlow);
+         var Q_brain = bloodFlow(BRAIN);
+         var Q_liver = bloodFlow(LIVER);
+         var Q_kidney = bloodFlow(KIDNEY);
+
+         var portal_factor = _portalFlowScalingFactor[score];
+         var liver_factor = _hepaticFlowScalingFactor[score];
+         var kidney_factor = _renalFlowScalingFactor[score];
+         var ci_factor = _cardiacIndexDiseaseFactor[score];
+
+         //update liver volume so that we get the correct diseases blood flow as Q_liver_HI = f(V_liver)
+         updateParameterFunc(new(volumeLiver, _hepaticVolumeScalingFactor[score]));
+
+         var Q_liver_HI = bloodFlow(LIVER);
+
+         //update all blood flows specs
+         portalOrgans.Each(x => updateBloodFlowSpec(x, portal_factor));
+         updateBloodFlowSpec(LIVER, liver_factor);
+         updateBloodFlowSpec(KIDNEY, kidney_factor);
+
+         //retrieve the scaling factor based on publication and github entry
+         //see https://github.com/Open-Systems-Pharmacology/Forum/discussions/1341
+         var otherOrganFactor = (ci_factor * (Q_other + Q_portal + Q_kidney + Q_liver + Q_brain) - (Q_brain + Q_portal * portal_factor + Q_liver_HI * liver_factor + Q_kidney * kidney_factor)) / Q_other;
```

Comment:
I am not sure if `Q_liver_HI` or `Q_liver `should be used in this equation - it depends on the definition of the blood flow `liver_factor` in the paper.
At the moment what happens in the code:
```
Volume_Liver_HI = volumeLiver*_hepaticVolumeScalingFactor[score];
QSpec_Liver_HI = QSpec_Liver*liver_factor;
```
So at the end we have:
`Q_Liver_HI = Q_Liver*_hepaticVolumeScalingFactor[score]*liver_factor`

So the overall scale factor for the liver bllod flow is `_hepaticVolumeScalingFactor[score]*liver_factor`
Is it correct?

Reply:
>So the overall scale factor for the liver blood flow is _hepaticVolumeScalingFactor[score]*liver_factor
Is it correct?

No this is not what the algorithm does. We only modify Q_spec_liv so this is ok 
What is weird to me is that the contribution of liver to calculate the impact on other factor seems to be accounted twice (via blood flow and volume)

Reply:
>
Qrest [L/min] = 1.21 + 0.29 + 1.33 = 2.83 (sum of healthy values of portal, hepatic arterial and renal blood flow in the individual that will be transformed to Child-Pugh A)
Qother [L/min] = 0.32 + 0.32 + 0.00325 + 0.26 + 1.12 + 0.33 = 2.35325 (sum of healthy values of “other organs” in the individual that will be transformed to Child-Pugh A)
Qbrain [L/min] = 0.78
(𝑄𝑟𝑒𝑠𝑡 × 𝐷𝐹_𝑟𝑒𝑠𝑡) [L/min] = 1.21 * 0.4 + 0.43 * 1.3 + 1.33 * 0.88 = 2.2134 (sum of Child-Pugh A adapted values of portal, hepatic arterial and renal blood flow)

this is from the githuib entry,. It's also using a modified valie of hepatic flow

Reply:
OK, just created an issue for this: https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/2621 
After reading of the paper I could not see if the "double scaling" of Q_liver is correct or not...

Reply:
Me neither. Ok so aside from this, this looks good? I'll make the renaming suggestions

Reply:
>Ok so aside from this, this looks good? I'll make the renaming suggestions

Just one last question regarding HCT-Scalefactor (s. last comment)

Reply:
Hematocrit value is another open point to address specifically. Those are absolute value that you see. Can you also create an issue for this?

Reply:
ok, issue created: https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/2622